share/completion/git-switch

```diff
@@ -0,0 +1,55 @@
+# (C) 2023 Victorien Elvinger
+
+# Completion script for the "git-switch" command.
+# Supports Git 2.23.0.
+
+function completion/git-switch {
+	WORDS=(git switch "${WORDS[2,-1]}")
+	command -f completion//reexecute
+}
+
+function completion/git::switch:arg {
+	OPTIONS=( #>#
+	"C: --force-create:; create or reset a new branch and switch to it"
+	"--conflict:; like --merge, but specify format of unmerged files"
+	"c: --create:; create a new branch and switch to it"
+	"d --detach; leave HEAD in detached head state"
+	"--guess; guess which branch to track based on its name"
+	"f --force; overwrite local changes or ignore unmerged files"
+	"--discard-changes; restore the index and worktree to match the destination branch"
+	"--ignore-other-worktrees; switch even if it is checked out by other worktrees"
+	"m --merge; do 3-way merge for destination branch"
+	"--no-guess; don't guess which branch to track based on its name"
+	"--no-progress; don't report progress status on standard error"
+	"--no-recurse-submodules; don't update the content of all active submodules"
+	"--no-track; create a non-tracking branch"
+	"--orphan:; create a new branch with no parent"
+	"--progress; report progress status on standard error"
+	"q --quiet; print error and warning messages only"
+	"--recurse-submodules; update the content of all active submodules"
+	"t --track; create a tracking branch"
+	) #<#
+
+	command -f completion//parseoptions -n
```

Comment:
When you are using `completion//getoperands`, you should not use the `-n` option for `completion//parseoptions`, so that `WORDS` are counted correctly. This affects completion of `git switch -c some_branch_name <TAB>`, where `-c` and `some_branch_name` are separate words.
```suggestion
	command -f completion//parseoptions
```

Reply:
I updated the completion to be more precise and helpful. The completion behavior now depends on the passed options.

Reply:
Fantastic!